app/utils/server/index.test.ts

```diff
@@ -0,0 +1,268 @@
+    alertServerAlreadyConnected,
+    alertServerError,
+    alertServerLogout,
+    alertServerRemove,
+    editServer,
+    isSupportedServer,
+    loginOptions,
+    loginToServer,
+    semverFromServerVersion,
+    sortServersByDisplayName,
+    unsupportedServer,
+} from './index';
+
+import type {ServersModel} from '@database/models/app';
+import type {DeepLinkWithData} from '@typings/launch';
+
+jest.mock('@utils/helpers', () => ({
+    isMinimumServerVersion: jest.fn(),
+    isTablet: jest.fn(() => true),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    dismissBottomSheet: jest.fn(),
+    showModal: jest.fn(),
+}));
+
+jest.mock('@components/compass_icon', () => ({
+    getImageSourceSync: jest.fn(),
+}));
+
+jest.mock('@utils/url', () => ({
+    tryOpenURL: jest.fn(),
+}));
+
+describe('unsupportedServer', () => {
+    const intl = getIntlShape();
+
+    it('should show the alert for sysadmin', () => {
+        const alert = jest.spyOn(Alert, 'alert');
+        unsupportedServer('Default Server', true, intl);
+        expect(alert?.mock?.calls?.[0]?.[2]?.length).toBe(2);
+    });
+
+    it('should not show the alert for team admin / user', () => {
+        const alert = jest.spyOn(Alert, 'alert');
+        unsupportedServer('Default Server', false, intl);
+        expect(alert).not.toHaveBeenCalled();
+    });
+});
+
+describe('isSupportedServer', () => {
+    it('should return true for supported server version', () => {
+        (isMinimumServerVersion as jest.Mock).mockReturnValue(true);
+        const result = isSupportedServer('5.0.0');
+        expect(result).toBe(true);
+    });
+
+    it('should return false for unsupported server version', () => {
+        (isMinimumServerVersion as jest.Mock).mockReturnValue(false);
+        const result = isSupportedServer('4.0.0');
+        expect(result).toBe(false);
+    });
+});
+
+describe('semverFromServerVersion', () => {
+    it('should return semver string for valid version', () => {
+        const result = semverFromServerVersion('5.0.0');
+        expect(result).toBe('5.0.0');
+    });
+
+    it('should return undefined for invalid version', () => {
+        const result = semverFromServerVersion('');
+        expect(result).toBeUndefined();
+    });
+});
+
+describe('addNewServer', () => {
+    it('should call dismissBottomSheet and showModal', async () => {
+        const theme = Preferences.THEMES.denim;
+        const serverUrl = 'https://server.com';
+        const displayName = 'Server';
+        const deepLinkProps = {type: 'channel', url: 'https://mattermost.com'} as DeepLinkWithData;
+
+        await addNewServer(theme, serverUrl, displayName, deepLinkProps);
+
+        expect(dismissBottomSheet).toHaveBeenCalled();
+        expect(showModal).toHaveBeenCalledWith(Screens.SERVER, '', expect.any(Object), expect.any(Object));
+    });
+});
+
+describe('loginOptions', () => {
+    it('should return correct login options', () => {
+        const config = {
```

Comment:
perhaps we should have a test helper that gives us back the base config and license objects.

Reply:
It hasn't come up too often so far when writing tests but definitely something we should consider if we find ourselves defining the config/license from scratch often